app/interactors/comments/destroy/find_record.rb

```diff
@@ -0,0 +1,19 @@
+module Comments
+  class Destroy
+    class FindRecord
+      include Interactor
+
+      delegate :id, to: :context
+
+      def call
+        comment = Comment.find_by(id: id)
+
+        if comment.nil?
+          context.fail!(errors: { base: ["Comment not found"] })
+        else
+          context.comment = comment
+        end
```

Comment:
Если не ошибаюсь при вызове context.fail! делается return, так что это должно сработать, также прочитай про blank? метод, предлагаю его использовать так как он более универсален

```suggestion
        context.fail!(errors: { base: ["Comment not found"] }) if comment.blank?

        context.comment = comment
```